tests/addons/environment.js

```diff
@@ -30,13 +29,36 @@ define([
 
     } else {
       this.requests = [];
+      this.responses = [];
       // switch to the fake XHR
       this.xhr = Sinon.useFakeXMLHttpRequest();
       this.xhr.onCreate = function (xhr) {
+        if (self.requests.length < self.responses.length) {
+          var mock = self.responses[self.requests.length];
+          setTimeout(function() {
+            xhr.respond(mock.status, mock.headers, mock.body);
+          }, 0);
+        }
```

Comment:
This is new, and handles the case where the mock response was set up before the XHR request was created.